### PR TITLE
fixed using gentoo live-dvd and added support for minimal install images

### PR DESCRIPTION
**Pull request by tismoe**
To make a long story short, I needed some bootable USB key that was EFI capable and would provide me a decent gentoo image. This didn't work out as expected as there was no support for the minimal installation image and the support for the live DVD image was expecting an erroneous file name (which may have been by intentional due to the distro-wise ambiguous naming scheme chosen by gentoo maintainers/ISO creators) and expected the image to contain both architectures, x86 as well as amd64.

Please have a look at the changes resulting from the pull request.

* gentoo DVD images were pulled using the wrong file name, this is fixed
* gentoo DVD images do not come in a mixed amd64/x86 flavour anymore, images are now considered platform specific
* support for gentoo minimal installation images has been added

### CODE REVIEW

**Comment by aguslr** [comment]
Hi @tismoe, thanks for taking the time to fix this.

> gentoo DVD images were pulled using the wrong file name, this is fixed

As you mentioned, that was intentional but I should probably add a note to the documentation regarding this distro. This should also apply to the minimal install ISO files.

> gentoo DVD images do not come in a mixed amd64/x86 flavour anymore, images are now considered platform specific

Are you referring to the minimal install? In the [download page for Gentoo](https://www.gentoo.org/downloads/), both the [LiveDVD amd64](http://bouncer.gentoo.org/fetch/gentoo-20160704-livedvd/amd64/) and the [LiveDVD x86](http://bouncer.gentoo.org/fetch/gentoo-20160704-livedvd/x86/) download links take you to a list of files that contain the ISO `livedvd-x86-amd64-32ul-20160704.iso` which is an amd64/x86 image. There is a multilib image for amd64 platforms but I see no exclusive x86 image. Am I missing something?

**Comment by tismoe** [comment]
Hi @aguslr, sorry for taking my time. As my original comment may have indicated, I was setting up a new development machine. I should be ready to go by now ;-).
> As you mentioned, that was intentional but I have added a note to the documentation regarding this distro with 826a264. This should also apply to the minimal install ISO files.

Absolutely makes sense. I'll stick to that convention.

> Are you referring to the minimal install? In the download page for Gentoo, both the LiveDVD amd64 and the LiveDVD x86 download links take you to a list of files that contain the ISO livedvd-x86-amd64-32ul-20160704.iso which is an amd64/x86 image. There is a multilib image for amd64 platforms but I see no exclusive x86 image. Am I missing something?

I think I sorted it out now. Both minimal install ISOs come with a single kernel they advertise in their boot menu. In fact, the `isolinux.cfg` of these two is identical. `install-minimal-generic.cfg` is therefore fine regarding the boot options.

When having a look at the live DVDs it turns out that the x86 ISO `livedvd-x86-amd64-32ul-20160704.iso` contains two kernels: one for plain x86 and one for amd64 both using the 32 bit environment of the ISO they come with as indicated by the description inside `isolinux.cfg`: "Boot Gentoo x86-amd64 in a 32ul environment. If you wish to have a 64ul then see livecd-amd64-multilib-*.iso to boot into a multilib 64ul environment".  The amd64 ISO comes with a single amd64 kernel instead which makes sense as booting and x86 kernel into an 64 bit environment would not make too much sense. 

**TL;DR** I would be going to split the `livedvd-generic.cfg` up, one for each live DVD ISO architecture provided in order to support the different boot options of these two. If you are okay with that please let me know and I'm on my way updating the PR.

**Comment by aguslr** [comment]
Hey @tismoe.

> I would be going to split the livedvd-generic.cfg up, one for each live DVD ISO architecture provided in order to support the different boot options of these two.

Yes, definitely I think it is best to have one config file (e.g. `livedvd-generic96.cfg`) for the ISO containing 32 + 64 bit kernels (`livedvd-x86-amd64-32ul-*.iso`) and another config file (maybe `livedvd-generic64.cfg`) for the multilib 64 bit only ISO (`livedvd-amd64-multilib-*.iso`). So please go ahead and update the PR.

Regarding the minimal install, what do you think about splitting it in two files (`minimal-generic32.cfg` and `minimal-generic64.cfg`) and load a file with the boot entry (`minimal-generic.inc`) since it's the same for both? This way we can check the platform before showing the entries. It's something similar to what we do what [BlackArch](https://github.com/aguslr/multibootusb/tree/master/grub.d/blackarch.d), [Manjaro](https://github.com/aguslr/multibootusb/tree/master/grub.d/manjaro.d), or [Void](https://github.com/aguslr/multibootusb/tree/master/grub.d/void.d).

So the file tree would be :

```
.
├── livedvd-generic64.cfg    # amd64-multilib entries
├── livedvd-generic96.cfg    # x86-amd64-32ul entries
├── minimal-generic32.cfg    # x86-minimal menu
├── minimal-generic64.cfg    # amd64-minimal menu
└── minimal-generic.inc      # x86-amd64-minimal common entry
```

What do you thing?

**Comment by aguslr** [comment]
> Yes, definitely I think it is best to have one config file (e.g. `livedvd-generic96.cfg`) for the ISO containing 32 + 64 bit kernels (`livedvd-x86-amd64-32ul-*.iso`) and another config file (maybe `livedvd-generic64.cfg`) for the multilib 64 bit only ISO (`livedvd-amd64-multilib-*.iso`). So please go ahead and update the PR.

> Regarding the minimal install, what do you think about splitting it in two files (`minimal-generic32.cfg` and `minimal-generic64.cfg`) and load a file with the boot entry (`minimal-generic.inc`) since it's the same for both? This way we can check the platform before showing the entries. It's something similar to what we do what [BlackArch](https://github.com/aguslr/multibootusb/tree/master/grub.d/blackarch.d), [Manjaro](https://github.com/aguslr/multibootusb/tree/master/grub.d/manjaro.d), or [Void](https://github.com/aguslr/multibootusb/tree/master/grub.d/void.d).

I'm going to go ahead and implement these changes myself.
